File: 2023-01/python/part1.py
def run(file):
    result = 0
    while line := file.readline():
        numbers = [int(c) for c in line if c.isdigit()]
        result += numbers[0] * 10 + numbers[-1]
    return result


if __name__ == "__main__":
    with open("../input.txt", 'r') as f:
        print(run(f))

File: 2023-01/rust/part1.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let filename = "../input.txt";
    let file = File::open(filename).unwrap();
    let mut result = 0;
    for line in BufReader::new(file).lines() {
        let numbers: Vec<u32> = line
            .unwrap()
            .chars()
            .filter_map(|c| c.to_digit(10))
            .collect();
        result += numbers.first().unwrap() * 10 + numbers.last().unwrap();
    }
    println!("{result}");
}
